Rakefile

```diff
@@ -28,6 +24,74 @@ namespace :yard do
     sh "bundle exec yard graph --protected --full --dependencies > #{output_file}"
     puts 'open doc/erd.dot if you have graphviz installed'
   end
+
+  # #files => Array<String>
+  # The Ruby source files (and any extra documentation files separated by '-') to process.
+  DOC_FILES = ['lib/**/*.rb', '-', 'README.md']
+  DOC_FILES.concat Rake::FileList['docs/**/*.md'].to_a
+  DOC_FILES.freeze
+  DOC_REF = `git log --pretty=format:'%h' -1 | less -F -X`.chomp
+  DOC_VERSION = ActiveModel::Serializer::VERSION
+  DOC_REMOTE = "git@github.com:rails-api/active_model_serializers.git"
+  DOC_REMOTE_NAME = 'origin'
+  DOC_ROOT = File.expand_path('..', __FILE__)
+  DOC_PATH = File.join('..', 'ams.doc')
+  DOC_DIR  = File.join(DOC_ROOT, DOC_PATH)
+
+  YARD::Rake::YardocTask.new(:doc) do |t|
+    t.stats_options = ['--list-undoc']
+    t.files = DOC_FILES
+  end
+
+  YARD::Rake::YardocTask.new(:pages) do |t|
+    t.files   = DOC_FILES
+    t.options = ['-o', DOC_PATH]
+  end
+
+  namespace :pages do
+    task :clean do
+      Dir.chdir(DOC_DIR) do
+        next unless File.directory?('.git')
+        sh 'git rm -rf . || echo $?' if system("git diff HEAD^..HEAD &>/dev/null | cat")
+      end
+    end
+
+    desc 'Check out gh-pages.'
+    task :checkout do
+      unless Dir.exist?(DOC_DIR)
+        Dir.mkdir(DOC_DIR)
+        Dir.chdir(DOC_DIR) do
+          sh 'git init'
+          sh "git remote add #{DOC_REMOTE_NAME} #{DOC_REMOTE}"
+          sh "git fetch #{DOC_REMOTE_NAME}"
+          if system('git branch -r gh-pages')
+            sh "cat .git/HEAD | grep -q 'gh-pages' || git checkout gh-pages || echo $?"
+            sh "git reset --hard #{DOC_REMOTE_NAME}/#{gh-pages}"
+          else
+            sh 'git checkout --orphan gh-pages'
+            sh 'git rm -rf . || echo $?'
+            sh 'echo "Page" > index.html'
+            sh 'git add index.html'
+            sh 'git commit -a -m "First pages commit"'
+            sh "git push #{DOC_REMOTE_NAME} gh-pages --force-with-lease"
+          end
+        end
+      end
+    end
+
+    desc 'Generate and publish YARD docs to GitHub pages.'
+    task publish: ['doc:pages:checkout', 'doc:pages:clean', 'doc:pages'] do
```

Comment:
Also see https://github.com/dbfit/dbfit/blob/f6bfc5147b9a37443ce6106bf06aa6ff6855e133/bin/publish-to-gh-pages.sh

```bash
#!/bin/bash
parent_sha=$(git show-ref -s refs/heads/gh-pages)
doc_sha=$(git ls-tree -d HEAD website | awk '{print $3}')
doc_commit_message=$(git log -n 1 --format="%s" $doc_dir)
new_commit=$(echo $doc_commit_message | git commit-tree $doc_sha -p $parent_sha)
git update-ref refs/heads/gh-pages $new_commit
git push origin gh-pages
```